google-cloud-storage/src/main/java/com/google/cloud/storage/NotificationInfo.java

```diff
@@ -0,0 +1,297 @@
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.pubsub.v1.ProjectTopicName;
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Google Storage Notification metadata;
+ *
+ * @see <a href="https://cloud.google.com/storage/docs/concepts-techniques#concepts">Concepts and
+ *     Terminology</a>
+ */
+public class NotificationInfo implements Serializable {
+
+  private static final long serialVersionUID = 5725883368559753810L;
+
+  public enum PayloadFormat {
+    JSON_API_V1,
+    NONE
+  }
+
+  static final Function<Notification, NotificationInfo> FROM_PB_FUNCTION =
+      new Function<Notification, NotificationInfo>() {
+        @Override
+        public NotificationInfo apply(Notification pb) {
+          return NotificationInfo.fromPb(pb);
+        }
+      };
+  static final Function<NotificationInfo, Notification> TO_PB_FUNCTION =
+      new Function<NotificationInfo, Notification>() {
+        @Override
+        public Notification apply(NotificationInfo NotificationInfo) {
+          return NotificationInfo.toPb();
+        }
+      };
+  private final String generatedId;
+  private final ProjectTopicName topic;
+  private final List<String> eventTypes;
+  private final Map<String, String> customAttributes;
+  private final PayloadFormat payloadFormat;
+  private final String objectNamePrefix;
+  private final String etag;
+  private final String selfLink;
+
+  public static final class Builder {
+
+    private String generatedId;
+    private ProjectTopicName topic;
+    private List<String> eventTypes;
+    private Map<String, String> customAttributes;
+    private PayloadFormat payloadFormat;
+    private String objectNamePrefix;
+    private String etag;
+    private String selfLink;
+
+    Builder(ProjectTopicName topic) {
+      this.topic = topic;
+    }
+
+    Builder(NotificationInfo NotificationInfo) {
+      generatedId = NotificationInfo.generatedId;
+      etag = NotificationInfo.etag;
+      selfLink = NotificationInfo.selfLink;
+      topic = NotificationInfo.topic;
+      eventTypes = NotificationInfo.eventTypes;
+      customAttributes = NotificationInfo.customAttributes;
+      payloadFormat = NotificationInfo.payloadFormat;
+      objectNamePrefix = NotificationInfo.objectNamePrefix;
+    }
+
+    Builder setGeneratedId(String generatedId) {
+      this.generatedId = generatedId;
+      return this;
+    }
+
+    Builder setSelfLink(String selfLink) {
+      this.selfLink = selfLink;
+      return this;
+    }
+
+    public Builder setTopic(ProjectTopicName topic) {
+      this.topic = topic;
+      return this;
+    }
+
+    public Builder setPayloadFormat(PayloadFormat payloadFormat) {
+      this.payloadFormat = payloadFormat;
+      return this;
+    }
+
+    /** GcpLaunchStage.Alpha */
```

Comment:
Please change this to either some meaningful description or remove it.

Reply:
done